File: HW7 (Collections)/LargeArray.js
function SubArray(size) {
	this.array = [];

	for(var i = 0; i < size; i++) {
		this.array[i] = 0;
	}		
}

function LargeArray(size) {
	this.array = [0,0,0];
	this.size = size
	this.chunks = Math.floor((size - 1) / 1000 + 1);

	for(var i = 0; i < this.chunks; i++) {
		if( i < this.chunks - 1 ) {
			this.array[i] = new SubArray(1000);
		} else {
			this.array[i] = new SubArray(size % 1000);
		}
	}

	this.set = function(position,value) {
		this.array[Math.floor(position / 1000)]
			.array[position % 1000] = value;
	}

	this.get = function(position) {
		return this.array[Math.floor(position / 1000)]
			.array[position % 1000];
	}
}

File: HW7 (Collections)/LargeArray.scala
class LargeArray(val size : Int) {
	val ars = new Array[Array[Int]](scala.math.ceil(size / 1000).toInt)
	var i = 0
	while( i < ars.length ) {
		if( i == ars.length - 1 ) {
			ars(i) = new Array[Int](size % 1000);
		} else {
			ars(i) = new Array[Int](1000);
		}
		i += 1
	}

	def get(i : Int) : Int = ars(i / 1000)(i % 1000)

	def set(i : Int, value : Int) {
		ars(i / 1000)(i % 1000) = value	
	}
}